theano/sandbox/gpuarray/opt.py

```diff
@@ -854,6 +858,44 @@ def values_eq_approx(a, b):
 register_opt()(conv_groupopt)
 
 
+# These two deal with any abstract convs that have a transfer somewhere
+@register_opt()
+@op_lifter([AbstractConv2d])
+def local_lift_abstractconv2d(node, context_name):
+    return [node.op(as_gpuarray_variable(node.inputs[0],
+                                         context_name=context_name),
+                    as_gpuarray_variable(node.inputs[0],
+                                         context_name=context_name))]
+
+@register_opt()
+@op_lifter([AbstractConv2d_gradWeights,
+            AbstractConv2d_gradInputs])
+def local_lift_abstractconv2dgrad(node, context_name):
+    return [node.op(as_gpuarray_variable(node.inputs[0],
+                                         context_name=context_name),
+                    as_gpuarray_variable(node.inputs[0],
+                                         context_name=context_name),
+                    node.inputs[2])]
+
+
+# This will deal with ops that don't have an explicit transfer but
+# have one of their inputs on the GPU already and the other not on the
+# GPU (to avoid endlessly replacing things).
+@register_opt()
+@local_optimizer()([AbstractConv2d])
```

Comment:
There is indeed an extra set of parentheses, but that should be legal syntax I think.

Anyway, I'll fix it.